compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -389,6 +389,60 @@ GatherOp::generateResultTileValue(OpBuilder &builder, unsigned resultNumber,
   return getTiledImplementation(builder, offsets, sizes);
 }
 
+LogicalResult GatherOp::generateScalarImplementation(OpBuilder &b, Location loc,
+                                                     ValueRange ivs) {
+  auto indexDepth = getIndexDepth();
+  Value result = b.create<memref::LoadOp>(loc, getOutput(), ivs);
+  SmallVector<Value> starts;
+  SmallVector<Value> loadIndices;
+  append_range(loadIndices, ivs.take_front(getBatchRank()));
+  //
+  // Populate with empty values.
+  auto sourceTy = getSourceType();
+  starts.resize(sourceTy.getRank(), Value());
+  auto resultIvs = ivs.drop_front(getBatchRank());
+  //
+  int64_t offset = starts.size() - resultIvs.size();
+  for (auto [idx, iv] : llvm::enumerate(resultIvs)) {
+    starts[idx + offset] = iv;
+  }
+  //
+  ArrayRef<int64_t> dimMap = getDimensionMap();
+  //
+  if (getIndicesType().getRank() > getBatchRank()) {
```

Comment:
I haven't looked at the semantics closely yet, but it is something like `bool hasBatch = `? I think it is clearer if we make it a variable and reuse it in the below code.